src/components/Avatar/styled/container.js

```diff
@@ -25,7 +25,7 @@ const StyledContainer = styled.span.attrs(props => {
 
     &:hover,
     &:focus {
-        color: currentColor;
+        color: ${props => props.getContrastText(props.brandMainColor)};
```

Comment:
since above we are using already:
```props.getContrastText(props.brandMainColor)}```
we should put this inside attrs function in order to compute getContrastText for brand color only once and inject a brandContrastTextColor like this:
```
const StyledContainer = styled.span.attrs(props => {
    const theme = getTheme(props);
    const { getContrastText, brand } = theme.palette;
    const brandMainColor = isValidColor(props.backgroundColor) ? props.backgroundColor : brand.main;
    const brandContrastTextColor = getContrastText(brandMainColor);
    return {
        brandMainColor,
        brandContrastTextColor,
    };
})
```